python/ray/tune/schedulers/pbt.py

```diff
@@ -231,14 +231,21 @@ def on_trial_add(self, trial_runner, trial):
         self._trial_state[trial] = PBTTrialState(trial)
 
     def on_trial_result(self, trial_runner, trial, result):
-        if self._time_attr not in result or self._metric not in result:
-            return TrialScheduler.CONTINUE
+        if self._time_attr not in result:
+            raise RuntimeError("Cannot find time_attr {} in trial results {}. "
```

Comment:
Hey, as mentioned previously, we should be cautious when doing this. The reason is because the user can return a result in tune.report that doesn't necessarily have the optimization metric.

To be backwards compatible, we should:

1. by default, raise RuntimeError if metric/time_attr is not found.
2. However, this should be flaggable (`"PBT(require_attrs=False)"`) which instead logs a warning message infrequently, using ray.utils.debug.log_once.

Reply:
> The reason is because the user can return a result in tune.report that doesn't necessarily have the optimization metric.

This will end up raising a KeyError anyways in line 249 (`result[self._metric]`). So this new change isn't actually changing current behavior but simply improving the error message.